EIPS/eip-custom_errors_for_erc_tokens.md

```diff
@@ -0,0 +1,393 @@
+
+### Parameter Glossary
+
+| Name        | Description                                                                         |
+| ----------- | ----------------------------------------------------------------------------------- |
+| `sender`    | The address of whose token(s) (is/are) transferred from.                            |
+| `balance`   | Current balance for the interacting account.                                        |
+| `needed`    | Amount minimum required to perform an action.                                       |
+| `receiver`  | Address owner of the token(s) transferred.                                          |
+| `spender`   | Address not owner of the token(s) transferred, but allowed to operate on (it/them). |
+| `allowance` | Amount of token(s) a `spender` is allowed to operate with.                          |
+| `approver`  | Owner of the token(s) being approved to an `spender`.                               |
+| `tokenId`   | The identifier number of a token type.                                              |
+| `owner`     | Address of the owner of a token type.                                               |
+| `operator`  | Same as `spender`.                                                                  |
+| `id`        | Same as `tokenId`.                                                                  |
+| `*Length`   | Array length for the prefixed parameter.                                            |
+
+### Error additions
+
+Any addition to this EIP is considerred a new proposal, but SHOULD follow the guidelines presented in the [rationale](#rationale) section to keep consistency.
+
+Implementation specific errors (such as extensions) SHOULD also follow the same guidelines, although subjects and arguments may vary.
+
+## Rationale
+
+The objectives of creating a standard for token errors are to provide context about the error, while keeping gramatical sense, and making a moderated use of meaningful arguments.
+
+Considering the outlined above, the errors are designed based on the standard actions that can be performed on each token, and the [subjects](#actions-and-subjects) involved.
+
+### Actions and subjects
+
+The main actions that can be performed within a token are:
+
+- **Transfer**: An operation in which a _sender_ moves any form of _balance_ to a _receiver_.
+- **Approval**: An operation in which an _approver_ grants any form of _approval_ to an _operator_.
+
+Whose subjects outlined are expected to be exhaustive to represent _what_ can go wrong in a transaction, which can be shown as an error by adding a [failure prefix](#error-prefixes).
+
+Note that `Token` or `TokenID` MUST NOT be a subject for the following reasons:
+
+1. The [Domain](#domain) is explicit enough to identify it as a token error.
+2. Although `TokenID` may identify an error, the token itself SHOULD NOT the root of the error, but can be additional information.
```

Comment:
I don't buy point 2 so strongly. I can imagine an ERC721 NFT with a supply of 10k tokens where transfers of token id > 10k result in an "InvalidTokenId" error. In fact, in this case what would be the error we'd use in OpenZeppelin's implementation? "InvalidOwner" isn't accurate because there is no "current owner" for an unminted token, address(0) is not considered a valid account so I wouldn't say it's the current owner. Maybe we need "InvalidTokenId" for this case!

If so, I would rephrase this part to say "The `Token` is never considered the subject of an error, but it is rather the context in which the error happens, as identified by the Domain."

Reply:
Section removed in favor of comments during our previous meeting.
Solving